Bio/SeqUtils/IsoelectricPoint.py

```diff
@@ -115,68 +116,34 @@ def _update_pKs_tables(self):
         return pos_pKs, neg_pKs
 
     # This function calculates the total charge of the protein at a given pH.
-    def _chargeR(self, pH):
+    # derivation:
+    #   Henderson Hasselbalch equation: pH = pKa + log([A-]/[HA])
+    #   Rearranging: [HA]/[A-] = 10 ** (pKa - pH)
+    #   partial_charge = [A-]/[A]total = [A-]/([A-] + [HA]) = 1 / { ([A-] + [HA])/[A-] } = 1 / (1 + [HA]/[A-]) = 1 / (1 + 10 ** (pKa - pH)) for acidic residues; 1 / (1 + 10 ** (pH - pKa)) for basic residues
+    def charge_at_pH(self, pH):
+        """Calculate the charge of a protein at given pH."""
         positive_charge = 0.0
         for aa, pK in self.pos_pKs.items():
-            CR = 10 ** (pK - pH)
-            partial_charge = CR / (CR + 1.0)
+            partial_charge = 1.0 / (10 ** (pH - pK) + 1.0)
             positive_charge += self.charged_aas_content[aa] * partial_charge
 
         negative_charge = 0.0
         for aa, pK in self.neg_pKs.items():
-            CR = 10 ** (pH - pK)
-            partial_charge = CR / (CR + 1.0)
+            partial_charge = 1.0 / (10 ** (pK - pH) + 1.0)
             negative_charge += self.charged_aas_content[aa] * partial_charge
 
         return positive_charge - negative_charge
 
-    def charge_at_pH(self, pH):
-        """Calculate the charge of a protein at given pH."""
-        return self._chargeR(pH)
-
     # This is the action function, it tries different pH until the charge of
     # the protein is 0 (or close).
-    def pi(self):
+    def pi(self, pH=7.775, min_=3.55, max_=12):
```

Comment:
Can you explain the default values? And extend the docstring accordingly?

Reply:
All solved.